src/particles/elements/ThinDipole.H

```diff
@@ -0,0 +1,120 @@
+/* Copyright 2022-2023 The Regents of the University of California, through Lawrence
+ *           Berkeley National Laboratory (subject to receipt of any required
+ *           approvals from the U.S. Dept. of Energy). All rights reserved.
+ *
+ * This file is part of ImpactX.
+ *
+ * Authors: Chad Mitchell, Axel Huebl
+ * License: BSD-3-Clause-LBNL
+ */
+#ifndef IMPACTX_THINDIPOLE_H
+#define IMPACTX_THINDIPOLE_H
+
+#include "particles/ImpactXParticleContainer.H"
+#include "mixin/beamoptic.H"
+#include "mixin/thin.H"
+#include "mixin/nofinalize.H"
+
+#include <AMReX_Extension.H>
+#include <AMReX_REAL.H>
+
+namespace impactx
+{
+    struct ThinDipole
+    : public elements::BeamOptic<ThinDipole>,
+      public elements::Thin,
+      public elements::NoFinalize
+    {
+        static constexpr auto name = "ThinDipole";
+        using PType = ImpactXParticleContainer::ParticleType;
+
+        static constexpr amrex::ParticleReal degree2rad = ablastr::constant::math::pi / 180.0;
+
+        /** A general thin-kick dipole element with chromatic effects
+         *
+         * This model is equivalent to the model described in Section 3.1 of
+         * G. Ripken, F. Schmidt, "A Symplectic Six-Dimensional Thin-Lens Formalism
+         * for Tracking," CERN/SL/95-12 (AP), 1995.  It is intended to replicate
+         * the thin-lens dipole model in MAD-X.
+         *
+         * @param theta - the total bending angle (degrees)
+         * @param rc - the curvature radius (m)
+         */
+        ThinDipole( amrex::ParticleReal const theta,
+                    amrex::ParticleReal const rc)
+        : m_theta(theta * degree2rad), m_rc(rc)
+        {
+        }
+
+        /** Push all particles */
+        using BeamOptic::operator();
+
+        /** This is a multipole functor, so that a variable of this type can be used like a
+         *  multipole function.
+         *
+         * @param p Particle AoS data for positions and cpu/id
+         * @param px particle momentum in x
+         * @param py particle momentum in y
+         * @param pt particle momentum in t
+         * @param refpart reference particle
+         */
+        AMREX_GPU_HOST_DEVICE AMREX_FORCE_INLINE
+        void operator() (
+                PType& AMREX_RESTRICT p,
+                amrex::ParticleReal & AMREX_RESTRICT px,
+                amrex::ParticleReal & AMREX_RESTRICT py,
+                amrex::ParticleReal & AMREX_RESTRICT pt,
+                [[maybe_unused]] RefPart const & refpart) const {
```

Comment:
The ref particle is used :)
```suggestion
                RefPart const & refpart) const {
```